leaderboard/service/get_around_me_test.go

```diff
@@ -0,0 +1,335 @@
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(membersFromService).To(Equal(membersReturn))
+		})
+
+		It("Should return error if getRank return member not found", func() {
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(-1, database.NewMemberNotFoundError(leaderboard, member))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewMemberNotFoundError(leaderboard, member)))
+		})
+
+		It("Should return error if getRank return in error", func() {
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(-1, fmt.Errorf("database error"))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
+		})
+
+		It("Should return error if TotalMembers return in error", func() {
+			rank := 6
+
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(rank, nil)
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(-1, fmt.Errorf("database error"))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
+		})
+
+		It("Should return error if GetOrderedMembers return in error", func() {
+			rank := 6
+			start, stop := calculateIndexes(totalMembers, pageSize, rank)
+
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(rank, nil)
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(totalMembers, nil)
+			mock.EXPECT().GetOrderedMembers(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(start), gomock.Eq(stop), gomock.Eq(order)).Return(nil, fmt.Errorf("database error"))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
+		})
+
+		It("Should ask for first 3 members if user is the first one", func() {
+			rank := 0
+			start := 0
+			stop := 2
+
+			mock.EXPECT().GetRank(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(member), gomock.Eq(order)).Return(rank, nil)
+			mock.EXPECT().GetTotalMembers(gomock.Any(), gomock.Eq(leaderboard)).Return(totalMembers, nil)
+
+			//this is the assertation relevant to this test
+			mock.EXPECT().GetOrderedMembers(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(start), gomock.Eq(stop), gomock.Eq(order)).Return(nil, fmt.Errorf("database error"))
+
+			_, err := svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
+			Expect(err).To(Equal(service.NewGeneralError("get around me", "database error")))
```

Comment:
It took me a while to understand what you wanted to test here.

You want to validate if the `GetAroundMe` calls the function with the parameters you specified. I think your assertion here is confusing and you can replace it with something more straightforward. Instead of validating the error and expecting the error, you are returning in your mock. Why not validate the number of times the function is called?

```suggestion
			//this is the assertation relevant to this test
			mock.EXPECT().GetOrderedMembers(gomock.Any(), gomock.Eq(leaderboard), gomock.Eq(start), gomock.Eq(stop), gomock.Eq(order)).Times(1)

			svc.GetAroundMe(context.Background(), leaderboard, pageSize, member, order, getLastIfNotFound)
```

`Times(1)` will validate if the method was called once with the arguments you specified